iquip/apps/scheduler.py

```diff
@@ -30,6 +34,40 @@ def _dismiss_items(layout: Optional[QLayout] = None):
                 _dismiss_items(item.layout())
 
 
+def _run_thread_with_worker(worker: QObject, parent: Optional[QObject] = None):
+    """Runs another thread with given worker.
+
+    Args:
+        worker: The worker that must be run through another thread. It must have:
+          - worker.run: the main function that has to be run.
+          - worker.done: the signal that is emitted when the work is done.
+    """
+    thread = QThread(parent=parent)
```

Comment:
Umm.. I think the `thread` will be removed just after finishing this function, then it may cause the error `destroyed a QThread object while running` you referred.

Is it possible to implement this function like [`explorer.py`](https://github.com/snu-quiqcl/iquip/blob/develop/iquip/apps/explorer.py#L118).